bookops_marc/bib.py

```diff
@@ -269,203 +236,137 @@ def main_entry(self) -> Optional[Field]:
+
+        # get OCLC #s from 991
+        subfield_991 = get_subvalue(self.get_fields("991"), "y")
+        if subfield_991 and self.library == "nypl":
+            unique_oclcs["991"] = subfield_991.without_prefix
 
         return unique_oclcs
 
-    def orders(self, sort: str = "descending") -> List[Order]:
+    @property
+    def orders(self) -> Sequence[Order]:
         """
-        Returns a list of order attached to bib
-
-        Args:
-            sort:                   ascending (from oldest to most recent) or
-                                    descending (from recent to oldest)
+        Returns a list of orders attached to bib. Order data coded in the 960 tag
+        (order fixed fields) may be followed by a related 961 tag (order variable
+        fields) so iterating over the entire bib is needed to connect the two
+        fields. It is possible that the 960 tag may not have a related 961 (BPL).
         """
-
-        if not isinstance(sort, str) or sort not in "ascending,descending":
-            raise BookopsMarcError("Invalid 'sort' argument was passed.")
-
         orders = []
 
-        # order data coded in the 960 tag (order fixed fields) may be followed by
-        # related 961 tag (order variable field) so iterating over entire bib
-        # is needed to connect these two;
-        # it is possible 960 tag may not have related 961 (BPL)
-
         for field in self:
             if field.tag == "960":
-                # shared NYPL & BPL mapping
-                oid = normalize_order_number(field.get(code="z"))
-
-                audns = self._get_shelf_audience_codes(field)
-                branches = self._get_branches(field)
-                copies = int(field.get(code="o"))
-                form = field.get(code="g")
-                created = normalize_date(field.get(code="q"))
-                lang = field.get(code="w")
-                shelves = self._get_shelves(field)
-                status = field.get(code="m").strip()
-
                 try:
-                    venNotes = None
                     following_field = self.fields[self.pos]
-                    if following_field.tag == "961":
-                        venNotes = following_field.get(code="h")
                 except IndexError:
-                    pass
-
-                o = Order(
-                    oid,
-                    audn=audns,
-                    branches=branches,
-                    copies=copies,
-                    created=created,
-                    form=form,
-                    lang=lang,
-                    shelves=shelves,
-                    status=status,
-                    venNotes=venNotes,
-                )
-                orders.append(o)
-
-        if sort == "descending":
-            orders.reverse()
-
+                    following_field = None
+                orders.append(Order(field, following_field))
         return orders
 
+    @property
     def overdrive_number(self) -> Optional[str]:
```

Comment:
I know this was not considered in the code before (quick hack), but it would be a good addition to the package:
It would be good if bookops-marc could identify correctly OverDrive ID in the 037s. Not every 037s will be related to OverDrive. You could use subfield $b Overdrive, Inc., etc. In other instances 037 can be ignored. This function should return only the first OverDrive ID (in the past we had more, but this was largely cleanup and it would be rather rare). Not sure is validating pattern of the ID is needed. Maybe we can rely on the presence and value of the subfield $b?